core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -655,15 +655,19 @@ private[spark] object Utils extends Logging {
       // created the directories already, and that they are secured so that only the
       // user has access to them.
       getYarnLocalDirs(conf).split(",")
+    } else if (conf.getenv("SPARK_LOCAL_DIRS") != null) {
```

Comment:
The origin complication was the shuffle service, but I think perhaps we should just keep it below as @vanzin mentioned since we added the check now. I also think that It's inconsistent to have the env variable take precendence where the spark.local.dir property doesn't.